CMakeLists.txt

```diff
@@ -33,40 +47,29 @@ else()
   set(hunter_config "")
 endif()
 
-set(package_deps)
-
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
     CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
   set(boost_components regex)
   hunter_add_package(Boost COMPONENTS regex)
-  list(APPEND LIBS Boost::regex)
```

Comment:
Does this break the Hunter build?

Reply:
No, if there's a library to link assocated with `Boost::regex` (for old compilers where it's not header-only, apparently) it's specified using `target_link_libraries` for `jaegertracing`.

Though come to think of it, that should be unconditional. On newer g++ the `Boost::regex` target will be an `INTERFACE` library instead.

Reply:
I'm not 100% sure why the build uses `hunter_add_package(Boost COMPONENTS regex)` for older gcc, and simply `hunter_add_package(Boost)` for newer. That seems ... weird. Wouldn't you just always ask for `Boost::regex` with `COMPONENTS regex` ?

Reply:
So the reason for that is the header-only vs. library dependencies in Boost. Hunter needs to know which libraries it is going to build. I'd rather not build or use the regex library from Boost if standard C++11 regex is available.